app/controllers/teams_controller.rb

```diff
@@ -0,0 +1,88 @@
+class TeamsController < ApplicationController
+  skip_before_filter :authenticate_user!, only: [:switch, :new, :create]
+  skip_after_filter :verify_authorized, only: [:switch, :new, :create]
+  skip_after_filter :verify_policy_scoped, only: [:switch, :new]
+
+  def switch
+    if current_user
+      team = current_user.teams.friendly.find(params[:id])
+      session[:current_team_slug] = team.slug
+    else
+      team = Team.not_archived.friendly.find(params[:id])
+      session[:team_slug] = team.slug
+    end
+    redirect_to root_path
+  end
+
+  # GET /teams/new
+  # GET /teams/new.xml
+  def new
+    @team = Team.new
+
+    respond_to do |format|
+      format.html # new.html.erb
+      format.xml  { render xml: @team }
+    end
+  end
+
+  # GET /teams/1/edit
+  def edit
+    @team = current_team
+    authorize @team
+  end
+
+  # POST /teams
+  # POST /teams.xml
+  def create
+    @team = Team.new(allowed_params)
+
+    respond_to do |format|
+      if verify_recaptcha && ( @team = TeamOperations::Create.(@team, nil) )
```

Comment:
Nope because it will just break the request. It does add error messages in the flash object to display in the form, so we can't break in the before_filter.